lib/rspec/rails/matchers/be_a_new.rb

```diff
@@ -49,7 +49,7 @@ def attributes
 
         def attributes_match?(actual)
           attributes.stringify_keys.all? do |key, value|
-            actual.attributes[key].eql?(value)
+            actual.attributes[key].eql?(value) || (value.respond_to?(:expected, true) && actual.attributes[key].include?(value.expected))
```

Comment:
One thing I've noticed upon further inspection is that while have_attributes does appear to use nested matchers the way you and the docs have said, there doesn't seem to actually be any testing around that in [the have_attributes_spec](https://github.com/rspec/rspec-expectations/blob/0ad53b7b0a6e34b1d20f64b1c6588a26648cf6ed/spec/rspec/matchers/built_in/have_attributes_spec.rb). There are a couple a_string_includings in there but those are just in "fail_with." It might be good to add some test coverage around that. Thoughts?